src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -669,6 +681,35 @@ private static void ValidateReadArgs(byte[] buffer, int offset, int count)
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) { throw new NotSupportedException(); }
             public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) { throw new NotSupportedException(); }
         }
+
+
+        private static void WriteToStream(Stream stream, string content, Encoding encoding)
+        {
+            if (ReferenceEquals(encoding, Encoding.UTF8) && content.Length <= 128)
+            {
+                // Fast-path for the expected common case
+                Span<byte> buffer = stackalloc byte[512];
+                int written = encoding.GetBytes(content, buffer);
+                stream.Write(buffer.Slice(0, written));
+            }
+            else
+            {
+                stream.Write(encoding.GetBytes(content));
+            }
+        }
+
+        private static void WriteLatin1ToStream(Stream stream, string content)
```

Comment:
Originally, we used `HttpRuleParser.DefaultHttpEncoding`, now we're using hardcoded `Latin1`. Is there a reason for the change?

Reply:
I did that here because the method explicitly calls for Latin1 in the name.

I don't believe we would ever change `HttpRuleParser.DefaultHttpEncoding` to something else due to compat, so I can use it here if we prefer it for consistency.

Reply:
That's what I'm asking :smile:  Why even introduce `WriteLatin1ToStream`?
I'm aware that the default HTTP encoding will hardly ever change, but the constant exists in our code, probably for a reason, and it's used in other places as well (`FormUrlEncodedContent`, `HeaderDescriptor`, `HttpConnection`).

Btw, the code looks the same as in `WriteToStream` hot path except the expected length array length (`IsSingleByte` on encoding could help with better stackalloc allocation)